staging/publishing/rules.yaml

```diff
@@ -643,6 +645,8 @@ rules:
     - repository: client-go
       branch: release-12.0
     - repository: code-generator
+      branch: release-1.15
+    - repository: component-base
```

Comment:
This doesn't really need to be added to 1.15 anymore. It leads to an extra unneeded sync but is a no-op so I'll clean up redundant rules in a follow-up. Example - https://github.com/kubernetes/kubectl/commit/f16387a692113404c05291b3d41ec50bf51af635